### DISCUSSION TITLE
with kairos, non-yum based OS get cumbersome even it was lightweight, why?

### DISCUSSION THREAD

**Discussion by liyimeng** [kairos-io/kairos#2262] [General]
when checking the size of the delivered iso files, I found an interesting pattern as below:

[kairos-alpine-3.19-core-amd64-generic-v3.0.0-alpha2.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-alpine-3.19-core-amd64-generic-v3.0.0-alpha2.iso) 1020 MB
[kairos-alpine-3.19-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-alpine-3.19-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso) 1.14 GB
[kairos-debian-bookworm-core-amd64-generic-v3.0.0-alpha2.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-debian-bookworm-core-amd64-generic-v3.0.0-alpha2.iso) 558 MB
[kairos-debian-bookworm-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-debian-bookworm-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso) 700 MB
[kairos-fedora-38-core-amd64-generic-v3.0.0-alpha2.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-fedora-38-core-amd64-generic-v3.0.0-alpha2.iso) 480 MB
[kairos-fedora-38-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-fedora-38-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso) 622 MB
[kairos-opensuse-leap-15.5-core-amd64-generic-v3.0.0-alpha2.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-opensuse-leap-15.5-core-amd64-generic-v3.0.0-alpha2.iso) 1020 MB
[kairos-opensuse-leap-15.5-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-opensuse-leap-15.5-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso) 1.13 GB
[kairos-rockylinux-9-core-amd64-generic-v3.0.0-alpha2.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-rockylinux-9-core-amd64-generic-v3.0.0-alpha2.iso) 438 MB
[kairos-rockylinux-9-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-rockylinux-9-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso) 580 MB
[kairos-ubuntu-20.04-core-amd64-generic-v3.0.0-alpha2.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-ubuntu-20.04-core-amd64-generic-v3.0.0-alpha2.iso) 858 MB
[kairos-ubuntu-20.04-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso](https://github.com/kairos-io/kairos/releases/download/v3.0.0-alpha2/kairos-ubuntu-20.04-standard-amd64-generic-v3.0.0-alpha2-k3sv1.27.9+k3s1.iso) 1000 MB

As you see, most of the lightweight os become a cumbersome one, why the heavy become much lighter, except the apt based package management. What could be the reason? because yum based system have dracut natively?

**Comment by Itxaka** [comment]
good point. Ususally the difference between core and standard is becuase we bundle k3s + deps with the standard so it makes sense that they are around 150Mb bigger.

But alpine should be smaller than that.

Usually the biggest files comes form the linux-firmware, which we fully bundle and it depends on the usptream on how many firmware files they bundle, usually its about 200 to 400 Mb of firmware files.

I'll have a look to see whats going on, the most obious ones form there are:

Alpine 1Gb -> should be much smaller
Leap 1Gb -> it also should be smaller

The rest seem ok to me, between 500Mb and 800Mb are to be expected. Any that you find they are disproportionately big?

**Reply by liyimeng** [reply]
Agree with you, can it be the reason that ubuntu-like system are  shipped  with a lot more firmware? Because the one that use ubuntu kernel are much bigger than those rhel variant equivalent.
